src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -185,14 +185,18 @@ export class FocusMonitor implements OnDestroy {
     let focusListener = (event: FocusEvent) => this._onFocus(event, nativeElement);
     let blurListener = (event: FocusEvent) => this._onBlur(event, nativeElement);
     this._ngZone.runOutsideAngular(() => {
-      nativeElement.addEventListener('focus', focusListener, true);
-      nativeElement.addEventListener('blur', blurListener, true);
+      // @ts-ignore
```

Comment:
Why are these `@ts-ignore` comments here?

Reply:
I suspect because they type is as a union `'focus' | 'blur' | 'click' | ...`, cast `'focusin' as any` is probably better than `@ts-ignore`

Reply:
`focusin` is a totally valid event for `addEventListener` (which falls back to `string` if for custom events)

Reply:
This is about the signature of the handler rather than the event type argument.

The `focusListener` and `blurListener` functions both take a `FocusEvent` as their argument - they have to, because they need to get at the `relatedTarget` property in order for this functionality to work (understanding where focus is going from _and_ to).

The typings in the version of typescript we are using call out the `focus` and `blur` events as producing a `FocusEvent`, but not `focusin` and `focusout` so they fall back to `Event`. This was incorrect ([per MDN](https://developer.mozilla.org/en-US/docs/Web/API/FocusEvent), all 4 events produce a `FocusEvent`) and has [since been fixed](https://github.com/microsoft/TypeScript/blame/e48cd3a1011ba0b61edee0cc19201cff8ff6f6e6/lib/lib.dom.d.ts#L5810), so the `@ts-ignore` is a temporary workaround.

Reply:
Ack, I would prefer to use `as any` with a TODO comment explaining this rather than `@ts-ignore` since the latter is a bigger hammer

Reply:
Okay I’ll make that change

Reply:
Reworked that bit and managed to avoid both `@ts-ignore` and `any`